### PR TITLE
Initialize variables and add macro for magic num

### PR DESCRIPTION
**Pull request by muneebahmed10**
*Description of changes:*
Initialize variables that gave "potentially uninitialized variables" warnings in visual studio, and define a macro for a magic number

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/mqtt/src/mqtt.c [hunk 1]
```diff
@@ -41,6 +41,11 @@
     #define MQTT_MAX_CONNACK_RECEIVE_RETRY_COUNT    ( 5U )
 #endif
 
+/**
+ * @brief A return code indicating an error from the transport interface.
+ */
+#define TRANSPORT_RETURNED_ERROR    ( -1 )
```
Comment (aggarw13) [inline libraries/standard/mqtt/src/mqtt.c:47]: Is the macro unused?

Comment (muneebahmed10) [reply libraries/standard/mqtt/src/mqtt.c:47]: Line 349

Comment (aggarg) [inline libraries/standard/mqtt/src/mqtt.c:47]: `TRANSPORT_ERROR` would probably be a better name?

File: libraries/standard/mqtt/src/mqtt.c [hunk 2]
```diff
@@ -340,8 +345,8 @@ static int32_t sendPacket( MQTTContext_t * pContext,
         }
         else
         {
-            LogError( ( "Transport send failed." ) );
-            totalBytesSent = -1;
+            LogError( ( "Transport send failed, BytesSent=%d.", bytesSent ) );
```
Comment (aggarg) [inline libraries/standard/mqtt/src/mqtt.c:348]:
```c
LogError( ( "Transport send failed. Error code=%d.", bytesSent ) );
```

File: libraries/standard/mqtt/src/mqtt.c (hunk 1 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/src/mqtt.c]: That's what I had originally, but then I changed it for some reason. Changing it back.

File: libraries/standard/mqtt/src/mqtt.c (hunk 2 above)
Comment (muneebahmed10) [reply libraries/standard/mqtt/src/mqtt.c]: Changed

**Pull request merged**